in-progress/8754-slashing-staking/design.md

```diff
@@ -0,0 +1,436 @@
+
+The staking contract have a few important tasks and requirements:
+
+- MUST provide a list of operators that the Rollup instance can use to sample from
+  - The list MUST have no duplicate `attester` to avoid annoying accounting for super-majority
+  - The list MUST have no gaps to avoid issues with sampling
+  - The list MAY contain duplicate `proposer`
+- MUST support adding a new operator to the list, given that they pass some permissionless anti-sybil mechanism
+- MUST support punishing an operator in the set if the misbehave
+- MUST support removing an operator from the list, with due time for punishing misbehaviour
+- SHOULD Minimise the impact to the operator, for operator **hot** keys being compromised
+- MAY be implemented directly on the Rollup instance
+
+> [!important] Notice
+> Notice that there are NO reference above to staking. Staking is a way to achieve a requirement but should not be a requirement on its own.
+
+## High level description
+
+Enrich the `Operator` description from the Rollup instance to include
+
+- `stake`
+- `status`
+- `withdrawer`
+
+We use **Proof-of-Stake** as an anti-sybil mechanism, requiring a stake of at least `MINIMUM_STAKE` to add an operator to the set.
+
+The `status` indicate whether the operator is currently `VALIDATING`, `LIVING` or `EXITING`:
+
+- `VALIDATING` meaning that it is currently active,
+- `LIVING` being that it is not participating in block production but also not exiting the system currently. It is possible to land in this state if a slash cause the validator to fall below the `MINIMUM_STAKE`.
+- `EXITING` the operator is currently waiting to exit or yet to claim the funds
+
+The `withdrawer` is a separate actor, specified per operator that is able to initialise a removal of the operator from the set - without the help of the `proposer` or `attester`. The `withdrawer` can initiate a withdrawal of the `stake` to some `recipient`, moving the status from `VALIDATING|LIVING` to `EXITING`, removing the operator from the set and start the delay. After the delay, the funds can be relayed to the specified `recipient`. This ensure that it is possible to keep the "account in control" in cold store or be a contract itself - lending itself well to supporting LST's
+
+We allow a `SLASHER` to reduce the `stake` of an operator, and updating its state to `LIVING` and removing it from the set if `stake < MINIMUM_STAKE`.
+
+The `SLASHER` can then be implemented as a variant of the `Governance.sol` contract, that skips the wider vote and allow direct execution of the payload if the validators agree. In this manner, the node software itself can be responsible for detecting misbehaviour and submitting slashing proposals, giving us a lot of flexibility as it is mainly off-chain code.
+
+> [!info] Top-ups  
+> We are not supporting "topping up" the account, since it introduce a few additional code paths that we simple are not interested in for this minimal setup.
```

Comment:
```suggestion
> We are not supporting "topping up" the account, since it introduce a few additional code paths that we simply are not interested in for this minimal setup.
```